apis/apiextensions/v1/composition_types.go

```diff
@@ -73,38 +74,44 @@ type CompositionSpec struct {
 	WriteConnectionSecretsToNamespace *string `json:"writeConnectionSecretsToNamespace,omitempty"`
 }
 
-// InlinePatchSets dereferences PatchSets and includes their patches inline. The
-// updated CompositionSpec should not be persisted to the API server.
-func (cs *CompositionSpec) InlinePatchSets() error {
+// ComposedTemplates processes and returns a Composition's resource templates,
+// defereferncing and inlining any PatchSets.
+func (cs *CompositionSpec) ComposedTemplates() ([]ComposedTemplate, error) {
```

Comment:
The idea here is to derive the merged templates from the Composition rather than sort of using the Composition as a 'transport' for them. Mostly just to avoid mutating the Composition and thus the (negligible) risk that we'd commit it back to the API server unintentionally.